internal/integration/cursor_test.go

```diff
@@ -304,74 +305,239 @@ func TestCursor(t *testing.T) {
 		batchSize = sizeVal.Int32()
 		assert.Equal(mt, int32(4), batchSize, "expected batchSize 4, got %v", batchSize)
 	})
+}
 
-	tailableAwaitDataCursorOpts := mtest.NewOptions().MinServerVersion("4.4").
-		Topologies(mtest.ReplicaSet, mtest.Sharded, mtest.LoadBalanced, mtest.Single)
+func parseMaxAwaitTime(mt *mtest.T, evt *event.CommandStartedEvent) int64 {
+	mt.Helper()
 
-	mt.RunOpts("tailable awaitData cursor", tailableAwaitDataCursorOpts, func(mt *mtest.T) {
-		mt.Run("apply remaining timeoutMS if less than maxAwaitTimeMS", func(mt *mtest.T) {
-			initCollection(mt, mt.Coll)
-			mt.ClearEvents()
+	maxTimeMSRaw, err := evt.Command.LookupErr("maxTimeMS")
+	require.NoError(mt, err)
 
-			// Create a find cursor
-			opts := options.Find().SetBatchSize(1).SetMaxAwaitTime(100 * time.Millisecond)
+	got, ok := maxTimeMSRaw.AsInt64OK()
+	require.True(mt, ok)
 
-			cursor, err := mt.Coll.Find(context.Background(), bson.D{}, opts)
-			require.NoError(mt, err)
+	return got
+}
 
-			_ = mt.GetStartedEvent() // Empty find from started list.
+func TestCursor_tailableAwaitData(t *testing.T) {
+	mt := mtest.New(t, mtest.NewOptions().CreateClient(false))
 
-			defer cursor.Close(context.Background())
+	// TODO(SERVER-96344): mongos doesn't honor a failpoint's full blockTimeMS.
+	mtOpts := mtest.NewOptions().MinServerVersion("4.4").
+		Topologies(mtest.ReplicaSet, mtest.LoadBalanced, mtest.Single)
+
+	mt.RunOpts("apply remaining timeoutMS if less than maxAwaitTimeMS", mtOpts, func(mt *mtest.T) {
+		cappedColl := newCappedCollection(mt, "tailable_awaitData_capped")
+		initCollection(mt, cappedColl)
```

Comment:
Optional: You can use `mtest.Options.CollectionCreateOptions` to create a capped collection.

Reply:
That is how the capped collection is being created here, see `newCappedCollection`.